lms/djangoapps/course_api/blocks/api.py

```diff
@@ -138,6 +138,10 @@ def get_blocks(
         for block_key in block_keys_to_remove:
             blocks.remove_block(block_key, keep_descendants=True)
 
+    # store transformed blocks in the current request to be reused where possible for optimization
+    if current_request := get_current_request():
+        setattr(current_request, "reusable_transformed_blocks", blocks)  # pylint: disable=literal-used-as-attribute
```

Comment:
Please use a `RequestCache` here, instead of setting this on the request object.

Reply:
@Serj-N: FWIW, I did a talk on caching in Django and in edx-platform. In case it comes in useful for later: https://www.youtube.com/watch?v=pa7igsOf4is